ts/components/ui/TabItem.tsx

```diff
@@ -0,0 +1,190 @@
+  Extrapolate,
+  interpolate,
+  interpolateColor,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withSpring
+} from "react-native-reanimated";
+import { WithTestID } from "../../types/WithTestID";
+import { IOIcons, Icon } from "../core/icons";
+import { HSpacer } from "../core/spacer/Spacer";
+import { LabelHeader } from "../core/typography/LabelHeader";
+import { IOScaleValues, IOSpringValues } from "../core/variables/IOAnimations";
+import { IOColors } from "../core/variables/IOColors";
+
+type ColorMode = "light" | "dark";
+
+type TabItem = WithTestID<{
+  label: string;
+  color?: ColorMode;
+  selected?: boolean;
+  // Icons
+  icon?: IOIcons;
+  iconSelected?: IOIcons;
+  // Accessibility
+  accessibilityLabel: string;
+  accessibilityHint?: string;
+  // Events
+  onPress: (event: GestureResponderEvent) => void;
+}>;
+
+type ColorStates = {
+  default: {
+    background: string;
+    foreground: IOColors;
+  };
+  selected: {
+    background: string;
+    foreground: IOColors;
+  };
+  pressed: {
+    background: string;
+  };
+};
+
+const mapColorStates: Record<NonNullable<TabItem["color"]>, ColorStates> = {
+  light: {
+    default: {
+      background: "transparent",
+      foreground: "grey-650"
+    },
+    selected: {
+      background: IOColors["grey-50"],
+      foreground: "black"
+    },
+    pressed: {
+      background: IOColors["grey-450"]
+    }
+  },
+  dark: {
+    default: {
+      background: "transparent",
+      foreground: "white"
+    },
+    selected: {
+      background: IOColors.white,
+      foreground: "grey-850"
+    },
+    pressed: {
+      background: IOColors["grey-100"]
+    }
+  }
+};
+
+const TabItem = ({
+  label,
+  color = "light",
+  selected = false,
+  accessibilityLabel,
+  accessibilityHint,
+  testID,
+  onPress,
+  icon,
+  iconSelected
+}: TabItem) => {
+  const isPressed: Animated.SharedValue<number> = useSharedValue(0);
+
+  // Scaling transformation applied when the button is pressed
+  const animationScaleValue = IOScaleValues?.basicButton?.pressedState;
+
+  // Using a spring-based animation for our interpolations
+  const progressPressed = useDerivedValue(() =>
+    withSpring(isPressed.value, IOSpringValues.button)
+  );
+
+  // Interpolate animation values from `isPressed` values
+  const pressedAnimationStyle = useAnimatedStyle(() => {
+    // Link color states to the pressed states
+
+    // Scale down button slightly when pressed
+    const scale = interpolate(
+      progressPressed.value,
+      [0, 1],
+      [1, animationScaleValue],
+      Extrapolate.CLAMP
+    );
+
+    return {
+      transform: [{ scale }]
+    };
+  });
+
+  const pressedColorAnimationStyle = useAnimatedStyle(() => {
+    // Link color states to the pressed states
+
+    const backgroundColor = interpolateColor(
+      progressPressed.value,
+      [0, 1],
+      [
+        mapColorStates[color].default.background,
+        `${mapColorStates[color].pressed.background}1A`
+      ]
+    );
```

Comment:
why not use one of the already defined hooks?

Reply:
Already defined hooks are too much specialized. But your comment made me think of a refactoring that could benefits different components.
In https://github.com/pagopa/io-app/pull/4822/commits/b27a2c9d086005e8d8d07300ded5b6bd6f711454 I made a small refactoring:
- Moved animation hooks in their own files, to make them more visible
- Changed their name to make clear what they are doing
- Extracted `pressProgress` shared value from hooks and made a dedicated hook named `usePressProgressValue`, so it's easier to use in other components.
- Used `usePressProgressValue` inside `TabItem` to make the custom animation (scale + bg)